cmd/restic/cmd_backup.go

```diff
@@ -175,12 +175,12 @@ func readLinesFromFile(filename string) ([]string, error) {
 
 // Check returns an error when an invalid combination of options was set.
 func (opts BackupOptions) Check(gopts GlobalOptions, args []string) error {
-	if opts.FilesFrom == "-" && gopts.password == "" {
+	if opts.FilesFrom != nil && opts.FilesFrom[0] == "-" && gopts.password == "" {
```

Comment:
Use `len(opts.FilesFrom) == 0` instead of the `nil` check: A slice may be non-nil and still have zero elements in it. Also, this only catches `--files-from -` when it is the first `--files-from` option... We need to loop over all elements of the slice and check for `-`.
```suggestion
	if gopts.password == "" {
		for _, filename := range opts.FilesFrom {
			if filename == "-" {
				return errors.Fatal("unable to read password from stdin when data is to be read from stdin, use --password-file or $RESTIC_PASSWORD")
			}
		}
	}
```

Reply:
This new review suggestion thing is cool!
Thanks for the info on nil/empty slices, I'm new to Go so I didn't know about that difference.

From what I can see the `--files-from -` behaviour is completely broken (with and without this patch), but that's another issue entirely.

Reply:
Yeah, we need to fix that, but it's a different issue so leave it for now.

Reply:
Hm, I had a look, and the `--files-from -` works for me as expected. What do you think is broken?